### ISSUE TITLE
An error occurred (BadRequestException) when calling the CreateDeployment operation: The REST API doesn't contain any methods

### ISSUE THREAD

**Issue by nguyenhoai890** [localstack/localstack#1909]
Hi team, 

I got new issue when trying using localstack@latest docker and localstack version 0.10.6. 

Run with command: `SLS_DEBUG=true sls deploy --stage=local`

localstack logs:
```
2019-12-31T08:22:22:WARNING:moto: No Moto CloudFormation support for AWS::S3::BucketPolicy
2019-12-31T08:22:27:WARNING:localstack.services.awslambda.lambda_api: Function not found: arn:aws:lambda:us-east-1:000000000000:function:publicspam-local-world
2019-12-31T08:22:28:INFO:localstack.services.cloudformation.cloudformation_starter: Updating resource ID from eefcf5b3 to pkqefv1g9n (us-east-1)
2019-12-31T08:22:28:WARNING:localstack.services.cloudformation.cloudformation_starter: Unable to determine physical_resource_id for resource <class 'moto.apigateway.models.RestAPI'>
2019-12-31T08:22:28:INFO:localstack.services.cloudformation.cloudformation_starter: Updating resource ID from 3a7ff272 to vxcggbi30s (us-east-1)
2019-12-31T08:22:28:WARNING:localstack.services.cloudformation.cloudformation_starter: Unable to determine physical_resource_id for resource <class 'moto.apigateway.models.Resource'>
2019-12-31T08:22:28:WARNING:localstack.utils.cloudformation.template_deployer: Error calling <bound method ClientCreator._create_api_method.<locals>._api_call of <botocore.client.APIGateway object at 0x7fdcab360e50>> with params: {'restApiId': 'pkqefv1g9n', 'stageName': 'local'} for resource: {'Type': 'AWS::ApiGateway::Deployment', 'Properties': {'RestApiId': 'pkqefv1g9n', 'StageName': 'local'}, 'DependsOn': ['ApiGatewayMethodWorldGet']}
2019-12-31T08:22:28:ERROR:localstack.services.cloudformation.cloudformation_starter: Unable to parse and create resource "ApiGatewayDeployment1577780537704": An error occurred (BadRequestException) when calling the CreateDeployment operation: The REST API doesn't contain any methods Traceback (most recent call last):
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_starter.py", line 178, in parse_and_create_resource
    return _parse_and_create_resource(logical_id, resource_json, resources_map, region_name)
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_starter.py", line 265, in _parse_and_create_resource
    result = deploy_func(logical_id, resource_wrapped, stack_name=stack_name)
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 723, in deploy_resource
    return execute_resource_action(resource_id, resources, stack_name, ACTION_CREATE)
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 749, in execute_resource_action
    result = configure_resource_via_sdk(resource_id, resources, resource_type, func, stack_name)
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 819, in configure_resource_via_sdk
    raise e
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 816, in configure_resource_via_sdk
    result = function(**params)
  File "/opt/code/localstack/.venv/lib/python3.7/site-packages/botocore/client.py", line 276, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/opt/code/localstack/.venv/lib/python3.7/site-packages/botocore/client.py", line 586, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.errorfactory.BadRequestException: An error occurred (BadRequestException) when calling the CreateDeployment operation: The REST API doesn't contain any methods

```
My env from sls:
```
Your Environment Information ---------------------------
     Operating System:          darwin
     Node Version:              12.13.1
     Framework Version:         1.58.0
     Plugin Version:            3.2.5
     SDK Version:               2.2.1
     Components Core Version:   1.1.2
     Components CLI Version:    1.4.0
```

Thanks,

**Comment by whummer** [comment]
Thanks for reporting @nguyenhoai890 . Can you please share a minimal `serverless.yml` configuration that helps reproduce this issue? Thanks

**Comment by nguyenhoai890** [comment]
> Thanks for reporting @nguyenhoai890 . Can you please share a minimal `serverless.yml` configuration that helps reproduce this issue? Thanks

Hi @whummer, srry for my late reply.
I just used the sample on the web:
```
service: demo

frameworkVersion: '>=1.28.0 <2.0.0'

plugins:
  - serverless-localstack

provider:
  name: aws
  runtime: go1.x

package:
  exclude:
    - ./**
  include:
    - ./bin/**

functions:
  world:
    handler: bin/world
    events:
      - http:
          path: world
          method: get
custom:
  localstack:
    debug: true
#    lambda:
#      mountCode: true
    stages:
      - local
    host: http://localhost
```

**Comment by chuckadams** [comment]
I have exactly the same problem.  It sometimes goes away when I restart localstack ... 20 or 30 times.  My serverless.yml

```
service: foo

provider:
  name: aws
  runtime: nodejs12.x

functions:
  hello:
    handler: handler.hello
    events:
      - http:
          path: hello
          method: post


plugins:
  - serverless-localstack

custom:
  localstack:
    stages:
      - local   
```

Not sure if it helps, but the word "hello" appears nowhere in `.serverless/cloudformation-template-create-stack.json`

**Comment by 4fthawaiian** [comment]
.. I'm seeing this as well. Would love to know if there's a workaround.

**Comment by whummer** [comment]
Thanks @nguyenhoai890 @chuckadams . I was able to reproduce this issue with the examples you provided. This should be fixed in #1998 - please re-open if this problem persists. Thanks

**Comment by micheleangioni** [comment]
Hi @whummer @nguyenhoai890 ,
I'm actually experiencing the same problem. I thought the Docker image would contain the fix.

I've created a simple repo to reproduce the problem (I am using MacOS):
https://github.com/micheleangioni/aws-nodejs-serverless-test

Is it something you can reproduce?
Thanks

**Comment by micheleangioni** [comment]
@whummer if it helps:

1. The problem doesn't happen 100% of times. More like 90%
2. The Docker image `0.10.3` seems not to be affected by the same problem

**Comment by micheleangioni** [comment]
@whummer other update: `0.10.4` and `0.10.5` seem NOT to be affected as well.

Furthermore, in the readme the `docker-compose.yaml` file is missing a line in the `volumes` section:

`- /var/run/docker.sock:/var/run/docker.sock`

**Comment by nguyenhoai890** [comment]
Hi @micheleangioni, I have just get the image docker localstack:latest and serverless-localstack and successfully started the lambda. 
I started with localstack start.

Update:
I have tried to deploy your source code. The only problem I can see is your env SLS_DEBUG. 
With your config: SLS_DEBUG = true. The deploy will fail since it checks that env should be string. So I changed to SLS_DEBUG: "*" and it's deployed successfully. 

```
service: aws-nodejs

provider:
  name: aws
  runtime: nodejs12.x
  stage: ${opt:stage, self:custom.defaultStage}
  environment:
    SLS_DEBUG: "*"
plugins:
  - serverless-localstack
```
